addons/event/models/event_mail.py

```diff
@@ -51,19 +53,18 @@ def _selection_template_model(self):
         ('before_event', 'Before the event'),
         ('after_event', 'After the event')],
         string='Trigger', default="before_event", required=True)
-    template_model_id = fields.Many2one('ir.model', string='Template Model', compute='_compute_template_model_id', compute_sudo=True)
-    template_ref = fields.Reference(string='Template', selection='_selection_template_model', required=True)
+    notification_type = fields.Selection([('mail', 'Mail')], string='Send', compute='_compute_notification_type')
+    template_ref = fields.Reference(string='Template', ondelete={'mail.template': 'cascade'}, required=True, selection=[('mail.template', 'Mail')])
 
-    @api.depends('notification_type')
-    def _compute_template_model_id(self):
-        mail_model = self.env['ir.model']._get('mail.template')
-        for mail in self:
-            mail.template_model_id = mail_model if mail.notification_type == 'mail' else False
+    @api.depends('template_ref')
+    def _compute_notification_type(self):
+        """Assigns the type of template in use, if any is set."""
+        sms_schedulers = self.filtered(lambda scheduler: scheduler.template_ref and scheduler.template_ref._name == 'mail.template')
+        sms_schedulers.notification_type = 'mail'
```

Comment:
"sms_scheduler" ? :smile: 

Would be easier to force "mail" on everything here, as it's the only available value.
And overrides would assign theirs.
That way you ensure having a value for every record, which is important for computed fields.